ert/storage/_storage.py

```diff
@@ -396,52 +396,42 @@ def _add_numerical_data(
         raise ert.exceptions.StorageError(meta_response.text)
 
 
+# the local variable dataframe is not read by pylint as pandas.Dataframe,
+# but due to chunking a pandas.TextFileReader
+# https://stackoverflow.com/questions/41844485/why-the-object-which-i-read-a-csv-file-using-pandas-from-is-textfilereader-obj
 def _response2records(
     response_content: bytes, metadata: _NumericalMetaData
 ) -> ert.data.RecordCollection:
-
+    assert metadata
     record_type = metadata.record_type
-
-    # the local variable dataframe is not read by pylint as pandas.Dataframe,
-    # but due to chunking a pandas.TextFileReader
-    # https://stackoverflow.com/questions/41844485/why-the-object-which-i-read-a-csv-file-using-pandas-from-is-textfilereader-obj
-
-    records: List[ert.data.Record]
-    if record_type == ert.data.RecordType.LIST_FLOAT:
-        dataframe = pd.read_csv(
-            io.BytesIO(response_content), index_col=0, float_precision="round_trip"
+    if record_type not in ert.data.RecordType.__members__:
+        raise ValueError(
+            f"Unexpected record type when loading numerical record: {record_type}"
         )
-        records = [
-            ert.data.NumericalRecord(data=row.to_list())
-            for _, row in dataframe.iterrows()  # pylint: disable=no-member
-        ]
-    elif record_type == ert.data.RecordType.MAPPING_INT_FLOAT:
-        dataframe = pd.read_csv(
-            io.BytesIO(response_content), index_col=0, float_precision="round_trip"
+
+    def interpret_data(row: pd.Series) -> Any:
```

Comment:
I think it would be better to pass the record type explicitly in the arguments and not rely on the `record_type` defined in the outer scope of this function.

Reply:
I had explicitly, like so:
`    def interpret_data(row: pd.Series) -> Any:
        if record_type == ert.data.RecordType.MAPPING_INT_FLOAT:
            return {int(k): v for k, v in row.to_dict().items()} 
       if record_type == ert.data.RecordType.MAPPING_STR_FLOAT:
           return row.to_dict()
       if record_type == ert.data.RecordType.LIST_FLOAT:
           return row.to_list()
`
but pylint does not like that. it prefer what I have now or the following:
`    def interpret_data(row: pd.Series) -> Any:
        if record_type == ert.data.RecordType.MAPPING_INT_FLOAT:
            return {int(k): v for k, v in row.to_dict().items()} 
       if record_type == ert.data.RecordType.MAPPING_STR_FLOAT:
           return row.to_dict()
      return row.to_list()
`
Should I settle for the second example for better readability?